secure-device-storage/sds-android/securedevicestorage/src/main/java/de/adorsys/android/securedevicestorage/SecurePreferences.java

```diff
@@ -24,52 +24,94 @@
 
 import static android.content.Context.MODE_PRIVATE;
 
+/**
+ * @author Drilon Reçica
+ * @since 2/17/17.
+ */
+
 public class SecurePreferences {
     private static final String KEY_SHARED_PREFERENCES_NAME = "SecurePreferences";
+    private static final String SALT_PREFIX = "SALT OF ";
+    private static final String KEY_CHARSET = "UTF-8";
 
     @RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
-    public static void setValue(@NonNull String key, @NonNull String value, @NonNull Context context) {
-        try {
-            if (!KeystoreTool.keyPairExists()) {
-                KeystoreTool.generateKeyPair(context);
+    public static void setValue(@NonNull String key, @NonNull String value,
+                                @NonNull Context context,
+                                @NonNull SecureMethod secureMethod) {
+        if (secureMethod.equals(SecureMethod.METHOD_ENCRYPT)) {
+            try {
+                if (!KeystoreTool.keyPairExists()) {
+                    KeystoreTool.generateKeyPair(context);
+                }
+            } catch (CertificateException | NoSuchAlgorithmException | IOException
+                    | KeyStoreException | UnrecoverableKeyException
+                    | InvalidAlgorithmParameterException | NoSuchProviderException e) {
+                Log.e(SecurePreferences.class.getName(), e.getMessage(), e);
```

Comment:
I think it's a good idea to throw a custom RuntimeException here so that the developer can get feedback that it didn't work